### PR TITLE
Feature/apply client side validation

### PR DESCRIPTION
**Pull request by arbulu89**
- Implementation of some client side validation patterns. This just raises errors if invalids values are used. The pattern description comes in the `extra_information` item
- Update sidebar menu texts

FYI: @petersatsuse

### CODE REVIEW

File: blue_horizon/azure/locales/custom-en.yml [hunk 1]
```diff
@@ -6,31 +6,39 @@ en:
...
   description: |
-    The solution will deploy the needed Azure infrastructure for SAP Netweaver and SAP HANA 
+    The solution will deploy the needed Azure infrastructure for SAP Netweaver and SAP HANA
     and provision and configure the virtual machines and install the SAP application for you.
-    
-    Doing SAP sizing right is not easy, therefore a simplified sizing approach is used. 
-    With the well known concept of T-Shirt sizes S,M,L we provide a simple way of using different settings. 
-    
+
+    Doing SAP sizing right is not easy, therefore a simplified sizing approach is used.
```
Comment (Simranpal) [inline blue_horizon/azure/locales/custom-en.yml:29]: `As SAP sizing can be complex, a simplified sizing approach is used.`

File: blue_horizon/azure/sources/variables.tf.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
     },
     "deployment_name": {
       "type": "string",
-      "description": "Suffix string added to some of the infrastructure resources names such as the resource group (rsg-ha-sap-${deployment_name})<!-- [group:Azure_configuration] -->"
+      "description": "Suffix string added to some of the infrastructure resources names such as the resource group (rsg-ha-sap-${deployment_name})<!-- [group:Azure_configuration] [pattern:/^[a-zA-Z0-9_.()-]{1,89}[^.]$/] [extra_information:Resource group names only allow alphanumeric characters, periods, underscores, hyphens and parenthesis and cannot end in a period] -->"
```
Comment (Simranpal) [inline blue_horizon/azure/sources/variables.tf.json:10]: cannot end` in/with` a period

File: blue_horizon/azure/sources/variables.tf.json [hunk 3]
```diff
@@ -32,26 +32,26 @@
     },
     "system_identifier": {
       "type": "string",
-      "default": "prd",
-      "description": "HANA SID. The System identifier (SID) must be 3 letters<!-- [group:HANA_configuration] -->"
+      "default": "PRD",
+      "description": "HANA SID. The System identifier (SID) must be 3 letters<!-- [group:HANA_configuration] [pattern:/^(?!ADD|ALL|AND|ANY|ASC|COM|DBA|END|EPS|FOR|GID|IBM|INT|KEY|LOG|MON|NIX|NOT|OFF|OMS|RAW|ROW|SAP|SET|SGA|SHG|SID|SQL|SYS|TMP|UID|USR|VAR)[A-Z][A-Z0-9]{2}$/] [extra_information:3 upper case characters string (the 2nd and 3rd can be digits). Next options are banned: ADD|ALL|AND|ANY|ASC|COM|DBA|END|EPS|FOR|GID|IBM|INT|KEY|LOG|MON|NIX|NOT|OFF|OMS|RAW|ROW|SAP|SET|SGA|SHG|SID|SQL|SYS|TMP|UID|USR|VAR] -->"
```
Comment (Simranpal) [inline blue_horizon/azure/sources/variables.tf.json:36]: `Next/Following` options are `banned/ not allowed`:

File: blue_horizon/azure/sources/variables.tf.json [hunk 4]
```diff
@@ -32,26 +32,26 @@
...
     "instance_number": {
       "type": "string",
       "default": "00",
-      "description": "HANA system instance number. Instance number must be 2 digits<!-- [group:HANA_configuration] -->"
+      "description": "HANA system instance number. Instance number must be 2 digits<!-- [group:HANA_configuration] [pattern:/^[0-9]{2}$/] [extra_information:2 digits string] -->"
     },
     "sap_admin_password": {
       "type": "string",
-      "description": "SAP administrator password. It is used to access to the sidadm user<!-- password_key [group:HANA_configuration] -->"
+      "description": "SAP administrator password. It is used to access to the sidadm user<!-- password_key [group:HANA_configuration] [pattern:/^[A-Za-z\d.]{1}(?=.*\d)(?=.*[a-z])(?=.*[A-Z])[A-Za-z\d]{7,}$/] [title:The password must contain, at least 8 characters, 1 digit, 1 upper-case character1 upper-case character, lower-case character and does not accept special characters]-->"
```
Comment (Simranpal) [inline blue_horizon/azure/sources/variables.tf.json:45]: `1 upper-case character1 upper-case character` is repeated

File: blue_horizon/azure/sources/variables.tf.json [hunk 5]
```diff
@@ -32,26 +32,26 @@
...
     },
     "deployment_type": {
       "type": "string",
       "default": "HA",
-      "description": "Select if the deployment creates a Cluster with HANA system replication and HA capabilities or not<!-- Options=[\"HA\",\"non HA\"] [group:HANA_configuration] -->"
+      "description": "Select if the deployment creates a Cluster with HANA system replication and HA capabilities or not<!-- Options=[HA,non HA] [group:HANA_configuration] -->"
```
Comment (Simranpal) [inline blue_horizon/azure/sources/variables.tf.json:50]: Select `whether` the deployment `should` create a Cluster with HANA system replication and HA capabilities or not.

**Review by Simranpal — COMMENTED** [review]
just some improvements to text below

File: blue_horizon/azure/sources/variables.tf.json (hunk 2 above)
Comment (arbulu89) [reply blue_horizon/azure/sources/variables.tf.json:10]: i just copied the text that Azure gives hehe

File: blue_horizon/azure/sources/variables.tf.json (hunk 5 above)
Comment (arbulu89) [reply blue_horizon/azure/sources/variables.tf.json:50]: I will only apply the `whether` change. I don't want to have conditional statements in the description

**Review by MalloZup — APPROVED** [review]
Sehr gut mein lieber :de:

**Review by Simranpal — APPROVED** [review]
thanks, looks good!

**Pull request merged**